### PR TITLE
[feature](function)support count_substrings functions

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes
support count_substrings functions
doc: https://github.com/apache/doris-website/pull/1211

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — DISMISSED** [review]
Please Add chinese TEST case

**Comment by zhangstar333** [comment]
> Please Add chinese TEST case

done

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

File: be/src/vec/functions/function_string.h [hunk 1]
```diff
@@ -2369,6 +2369,124 @@ class FunctionSplitByString : public IFunction {
...
+class FunctionCountSubString : public IFunction {
+public:
+    static constexpr auto name = "count_substrings";
+
+    static FunctionPtr create() { return std::make_shared<FunctionCountSubString>(); }
+    using NullMapType = PaddedPODArray<UInt8>;
+
+    String get_name() const override { return name; }
+
+    bool is_variadic() const override { return false; }
```
Comment (HappenLee) [inline be/src/vec/functions/function_string.h:2381]: unless code delete

File: be/src/vec/functions/function_string.h [hunk 2]
```diff
@@ -2369,6 +2369,124 @@ class FunctionSplitByString : public IFunction {
...
+        for (size_t i = 0; i < input_rows_count; ++i) {
+            const StringRef pattern_ref = pattern_col.get_data_at(i);
+            dest_column_data[i] = find_str_count(str_ref, pattern_ref);
+        }
+    }
+
+    size_t find_pos(size_t pos, const StringRef str_ref, const StringRef pattern_ref) const {
+        size_t old_size = pos;
+        size_t str_size = str_ref.size;
+        while (pos < str_size && memcmp_small_allow_overflow15(str_ref.data + pos, pattern_ref.data,
```
Comment (HappenLee) [inline be/src/vec/functions/function_string.h:2465]: how ck impl ？ it's effective way to do the work?

Comment (zhangstar333) [reply be/src/vec/functions/function_string.h:2465]: Almost the same

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**